app/_src/guides/gateway-api.md

```diff
@@ -0,0 +1,336 @@
+---
+title: Kubernetes Gateway API
+---
+
+To get traffic from outside your mesh inside it (North/South) with {{site.mesh_product_name}} you can use
+a builtin gateway.
+
+In the [quickstart](/docs/{{ page.version }}/quickstart/kubernetes-demo/), traffic was only able to get in the mesh by port-forwarding to an instance of an app
+inside the mesh.
+In production, you typically set up a gateway to receive traffic external to the mesh.
+In this guide you will add [a built-in gateway](/docs/{{ page.version }}/using-mesh/managing-ingress-traffic/builtin/) in front of the demo-app service and expose it publicly.
+We will deploy and configure Gateway using [Kubernetes Gateway API](https://gateway-api.sigs.k8s.io/).
+
+Service graph of the demo app with a builtin gateway on front:
+{% mermaid %}
+flowchart LR
+  subgraph edge-gateway
+    gw0(/ :8080)
+  end
+  demo-app(demo-app :5000)
+  redis(redis :6379)
+  gw0 --> demo-app
+  demo-app --> redis
+{% endmermaid %}
+
+## Prerequisites
+
+- Completed [quickstart](/docs/{{ page.version }}/quickstart/kubernetes-demo/) to set up a zone control plane with demo application
+
+## Install Gateway API CRDs
+
+To install Gateway API please refer to [official installation instruction](https://gateway-api.sigs.k8s.io/guides/#install-standard-channel).
+
+You also need to manually install {{site.mesh_product_name}} [GatewayClass](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/):
+
+```shell
+echo "apiVersion: gateway.networking.k8s.io/v1
+kind: GatewayClass
+metadata:
+  name: kuma
+spec:
+  controllerName: gateways.kuma.io/controller" | kubectl install -f -
+```
+
+At this moment, when you install Gateway API CRDs after installing {{site.mesh_product_name}} control plane you need to restart
+it to start Gateway API controller. To do this run: 
+
+```shell
+kubectl rollout restart deployment {{site.mesh_product_name_path}}-control-plane -n {{ site.mesh_namespace }}
+```
+
+## Start a gateway
+
+The [Gateway](https://gateway-api.sigs.k8s.io/api-types/gateway/) resource represents the proxy instance that handles
+traffic for a set of Gateway API routes. You can create gateway with a single listener on port 8080 by running:
+
+```shell
+echo" apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: kuma
+  namespace: kuma-demo
+spec:
+  gatewayClassName: kuma
+  listeners:
+   - name: proxy
+     port: 8080
+     protocol: HTTP" | kubectl apply -f -
+```
+
+{% warning %}
+The Kubernetes cluster needs to support `LoadBalancer` for this to work.
```

Comment:
alternatively we can could create a GatewayClass + MeshGatewayConfig that uses ClusterIP

Reply:
do we need an issue for this? If yes, can you create one with better explanation? I don't want to miss any context

Reply:
Probably this is good as is